### PR TITLE
pybind11: correct get_include path

### LINKED ISSUE #13559: pybind11: fix get_include()
The `py-pybind11` package has a python module helper of the form:

```python
import pybind11 as py
print(py.get_include())
```

which helps downstream, if they are not using the CMake config package (which probably everyone is using), to query the install location.

`get_include()` currently points to the spec's Python include location instead of the package's include.

Diff for a new unit test:
```diff
diff --git a/var/spack/repos/builtin/packages/py-pybind11/package.py b/var/spack/repos/builtin/packages/py-pybind11/package.py
index cd482ceb3..0a86afb97 100644
--- a/var/spack/repos/builtin/packages/py-pybind11/package.py
+++ b/var/spack/repos/builtin/packages/py-pybind11/package.py
@@ -60,3 +60,17 @@ class PyPybind11(CMakePackage):
         super(PyPybind11, self).install(spec, prefix)
         setup_py('install', '--single-version-externally-managed', '--root=/',
                  '--prefix={0}'.format(prefix))
+
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    def test(self):
+        with working_dir('spack-test', create=True):
+            # test include helper points to right location
+            module = "pybind11"
+            python = Executable(self.spec['python'].command.path)
+            python('-c', 'import {0}'.format(module))
+            inc = python('-c', 'import {0} as py; print(py.get_include())'.format(
+                module), output=str)
+            print("INC:")
+            print(inc)
+            assert inc == self.prefix.include
```

Similar to:
- https://github.com/conda-forge/pybind11-feedstock/pull/32
- https://github.com/pybind/pybind11/pull/1877

### PR DESCRIPTION
**Pull request by ax3l**
Helper for non-CMake downstream projects to find the pybind11 header location.

Patch the right location of our install prefix in.

Fix #13559

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pybind11/package.py [hunk 1]
```diff
@@ -56,7 +56,25 @@ def cmake_args(self):
...
         super(PyPybind11, self).install(spec, prefix)
         setup_py('install', '--single-version-externally-managed', '--root=/',
                  '--prefix={0}'.format(prefix))
+
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    def test(self):
+        with working_dir('spack-test', create=True):
+            # test include helper points to right location
+            python = Executable(self.spec['python'].command.path)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybind11/package.py]:
```suggestion
            python = self.spec['python'].command
```

File: var/spack/repos/builtin/packages/py-pybind11/package.py [hunk 2]
```diff
@@ -56,7 +56,25 @@ def cmake_args(self):
...
+            inc = python('-c',
+                         'import pybind11 as py; print(py.get_include())',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybind11/package.py]: This print syntax will only work in Python 2.6+. You should be able to do something like:
```python
'import pybind11 as py; ' + self.spec['python'].print_string('py.get_include()')
```

File: var/spack/repos/builtin/packages/py-pybind11/package.py (hunk 1 above)
Comment (ax3l) [reply var/spack/repos/builtin/packages/py-pybind11/package.py]: Ah cool, thanks!

File: var/spack/repos/builtin/packages/py-pybind11/package.py (hunk 2 above)
Comment (ax3l) [reply var/spack/repos/builtin/packages/py-pybind11/package.py]: That's okay, pybind11 only supports Python 2.7+ ;)
https://github.com/pybind/pybind11/#goodies

**Pull request merged**